docs/developer-docs/latest/development/custom-fields.md

```diff
@@ -0,0 +1,340 @@
+
+| Component | Description                                                         |
+| --------- | ------------------------------------------------------------------- |
+| `Input`   | React component to use in the Content Manager's edit view           |
+| `View`    | Read-only React component to use in the Content Manager's list view |
+
+::: details Example: Registering Input and View components imported from dedicated files:
+
+<!-- ? can we declare components inline as well? like Input: async () => <MyReactComponent>…<MyReactComponent/>? What would be the most simple React.component example we could describe for a custom field? -->
+```js
+// path: ./src/plugins/my-custom-field-plugin/strapi-admin.js
+
+register(app) {
+  app.customFields.register({
+    // …
+    components: {
+      Input: async () => import(/* webpackChunkName: "input-component" */ "./Input"),
+      View: async () => import(/* webpackChunkName: "view-component" */ "./View"),
+    } 
+    // …
+  });
+}
+```
+
+:::
+
+### Options
+
+`app.customFields.register()` can pass an additional `options` object with the following parameters:
+
+| Options parameter | Description                                                                     | Type                    |
+| -------------- | ------------------------------------------------------------------------------- | ----------------------- |
+| `base`         | Settings available in the _Base settings_ tab of the field in the Content-type Builder       | `Object` or  `Array of Objects` |
+| `advanced`     | Settings available in the _Advanced settings_ tab of the field in the Content-type Builder   | `Object` or  `Array of Objects` |
+| `validator`    | Validator function returning an object, used to sanitize input | `Function`              |
+
+Both `base` and `advanced` settings accept an object or an array of objects, each object being a settings section. Each settings section must include:
+
+- a `sectionTitle` to declare the title of the section as an [`IntlObject`](https://formatjs.io/docs/react-intl/)
+- and a list of `items` as an array of objects.
+
+Each object in the `items` array can contain the following parameters:
+
+<!-- TODO: fix the table content as the parameters given in the [example from the RFC](https://github.com/strapi/rfcs/blob/3eb034e169746558315d719ca5fb49cec854640a/rfcs/xxxx-custom-fields-api.md#example) and the [TS interface](https://github.com/strapi/rfcs/blob/3eb034e169746558315d719ca5fb49cec854640a/rfcs/xxxx-custom-fields-api.md#admin) described in the RFC are inconsistent 🤷  -->
+
+| Items parameter | Description                                                                                  | Type     |
+| --------------- | -------------------------------------------------------------------------------------------- | -------- |
+| `name`          | Name of the setting item<br/>Must use the `options.setting-name` format.                     | `String` |
```

Comment:
I think we are talking about inputs for the settings here if that makes sense, so the name of the input (technically in terms of html semantics I think it is the label)

Reply:
I believe it should be camelCase (`settingName`), since it gets written to `schema.json` files, and everything there is camelCase